vmm/vmm.go

```diff
@@ -0,0 +1,119 @@
+package vmm
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"sync"
+
+	"github.com/bobuhiro11/gokvm/flag"
+	"github.com/bobuhiro11/gokvm/machine"
+	"github.com/bobuhiro11/gokvm/term"
+)
+
+type BootConfig struct {
+	KernFile   *os.File
+	InitrdFile *os.File
+	flag.Config
+}
+
+type VMM struct {
+	*machine.Machine
+	BootConfig
+}
+
+func New(c flag.Config) *VMM {
+	return &VMM{
+		Machine: nil,
+		BootConfig: BootConfig{
+			Config: c,
+		},
+	}
+}
+
+// Init instantiates a machine.
+func (v *VMM) Init() error {
+	m, err := machine.New(v.Dev, v.NCPUs, v.MemSize)
+	if err != nil {
+		return err
+	}
+
+	if len(v.TapIfName) > 0 {
+		if err := m.AddTapIf(v.TapIfName); err != nil {
+			return err
+		}
+	}
+
+	if len(v.Disk) > 0 {
+		if err := m.AddDisk(v.Disk); err != nil {
+			return err
+		}
+	}
+
+	kern, err := os.Open(v.Kernel)
+	if err != nil {
+		return err
+	}
+
+	initrd, err := os.Open(v.Initrd)
+	if err != nil {
+		return err
+	}
```

Comment:
Since vmm.Setup() does the loading using the various boot protocols, it would be better to move these code pieces to vmm.Setup(). What do you think?

Reply:
done